### PR TITLE
feat: Add `table_name` to `Partition` response in management client

### PR DESCRIPTION
**Pull request by alamb**
# Rationale:

I want to be able to call `management::Client::compact_object_store_chunks` as part of https://github.com/influxdata/influxdb_iox/issues/3412:

```rust
pub async fn compact_object_store_chunks(
    &mut self,
    db_name: impl Into<String> + Send,
    table_name: impl Into<String> + Send,
    partition_key: impl Into<String> + Send,
    chunk_ids: Vec<Bytes>
) -> Result<IoxOperation, Error>
...
```

However, to do so you I must supply the `table_name` which is not returned in the call to `list_partitions` or `get_partition`

# Changes:
1. Add `table_name` to `Partition` in management protobuf
2. Thread `PartitionAddr` around enough to get the info

# Aside
I can't help thinking the access of Partiton info / Chunk info could be more uniform but for now, this seems ok

### CODE REVIEW

File: data_types/src/partition_metadata.rs [hunk 1]
```diff
@@ -59,13 +59,6 @@ impl PartitionSummary {
     }
 }
 
-/// Metadata and statistics for a Chunk *within* a partition
```
Comment (alamb) [inline data_types/src/partition_metadata.rs:62]: drive by cleanup -- this structure is not used

File: generated_types/protos/influxdata/iox/management/v1/partition.proto [hunk 2]
```diff
@@ -9,4 +9,7 @@ option go_package = "github.com/influxdata/iox/management/v1";
 message Partition {
   // The partition key of this partition
   string key = 1;
+
```
Comment (alamb) [inline generated_types/protos/influxdata/iox/management/v1/partition.proto:12]: The point of this PR is to add this field and fill it correctly in the management response

File: query_tests/src/scenarios.rs [hunk 3]
```diff
@@ -676,12 +676,13 @@ impl DbSetup for TwoMeasurementsManyFieldsTwoChunks {
         ];
         write_lp(&db, &lp_lines.join("\n"));
 
-        assert_eq!(count_mutable_buffer_chunks(&db), 2);
-        assert_eq!(count_read_buffer_chunks(&db), 1);
+        assert_eq!(count_mutable_buffer_chunks(&db), 4);
```
Comment (alamb) [inline query_tests/src/scenarios.rs:679]: This db is partitioned on `table_name` and there are two tables (each with 2 MUbs and a rub). Previously the code grouped on partition_key (which doesn't have the table name) so it was off by a factor of 2

**Review by carols10cents — APPROVED** [review]
LGTM!

**Pull request merged**